start.php

```diff
@@ -33,12 +30,14 @@ function mentions_get_regex() {
 	// @todo this won't work for usernames that must be html encoded.
 	// get all chars with unicode 'letter' or 'mark' properties or a number _ or .,
 	// preceeded by @, and possibly surrounded by word boundaries.
-	return '/[\b]?@([\p{L}\p{M}_\.0-9]+)[\b]?/iu';
+	return '/[\b]?@([\p{L}\p{M}-_\.0-9]+)[\b]?/iu';
 }
 
 function mentions_get_views() {
 	// allow plugins to add additional views to be processed for usernames
 	$views = array('output/longtext');
+	//, 'river/item');
+	//,'object/thewire');
```

Comment:
i was in the process of extending the application of the mentions plugin to apply to every location in elgg where it makes sense to have mentions visible, such as river items and wire posts. however, i found that it was interfered with by wire_tools on my installation so i commented them out. i didn't fully prepare the code for a straight integration into the parent version of mentions before creating the pull request. if you'd prefer me to do that then i will have to cancel this one and do it again in a few days.